src/modules/core.js

```diff
@@ -651,6 +651,10 @@
 					request.minor = 'true';
 				}
 
+				if ( ['nochange', 'unwatch', 'watch'].includes(options.watchlist) ) {
```

Comment:
Why is this preference being used here ? This in most cases does not trigger watchlisting unless the user explicitly watchlists all edits

Reply:
Because many users do opt to "Add pages and files I edit to my watchlist" since the mediawiki preferences don't have the granularity to specify things "only add all talk pages I edit" or "don't add tool edits". I'll admit that this line is more complicated than it necessarily needs to be at this point (this could be a boolean option between `nochange` and `preferences`), but I tend to prefer adding more flexibility to my code to allow easier expansion in the future.